Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs

```diff
@@ -8,230 +8,183 @@
+#if UNITY_2023_2_OR_NEWER
+            reporter ??= new DefaultInputActionAssetValidationReporter();
+            CheckForDefaultUIActionMapChanges(asset, reporter);
+#endif // UNITY_2023_2_OR_NEWER
+            return true;
         }
 
-        /// <summary>
-        /// Delete project wide input actions
-        /// </summary>
-        internal static void DeleteActionAssetAndActionReferences()
+        internal static bool ValidateAndSaveAsset(InputActionAsset asset, IReportInputActionAssetValidationErrors reporter = null)
         {
-            var objects = AssetDatabase.LoadAllAssetsAtPath(s_AssetPath);
-            if (objects != null)
-            {
-                // Handle deleting all InputActionAssets as older 1.8.0 pre release could create more than one project wide input asset in the file
-                foreach (var obj in objects)
-                {
-                    if (obj is InputActionReference)
-                    {
-                        var actionReference = obj as InputActionReference;
-                        actionReference.Set(null);
-                        AssetDatabase.RemoveObjectFromAsset(obj);
-                    }
-                    else if (obj is InputActionAsset)
-                    {
-                        AssetDatabase.RemoveObjectFromAsset(obj);
-                    }
-                }
-            }
+            Validate(asset, reporter); // Currently ignoring validation result
+            return InputActionAssetManager.SaveAsset(asset);
+        }
+
+        private static bool ReportError(IReportInputActionAssetValidationErrors reporter, InputAction action, string message)
+        {
+            return reporter.OnValidationError(action, message);
         }
 
+#if UNITY_2023_2_OR_NEWER
         /// <summary>
-        /// Updates the input action references in the asset by updating names, removing dangling references
-        /// and adding new ones.
+        /// Checks if the default UI action map has been modified or removed, to let the user know if their changes will
+        /// break the UI input at runtime, when using the UI Toolkit.
         /// </summary>
-        internal static void UpdateInputActionReferences()
+        internal static bool CheckForDefaultUIActionMapChanges(InputActionAsset asset, IReportInputActionAssetValidationErrors reporter = null)
         {
-            var asset = GetOrCreate();
-            var existingReferences = InputActionImporter.LoadInputActionReferencesFromAsset(asset).ToList();
+            reporter ??= new DefaultInputActionAssetValidationReporter();
+
+            var defaultUIActionMap = GetDefaultUIActionMap();
+            var uiMapIndex = asset.actionMaps.IndexOf(x => x.name == "UI");
 
-            // Check if referenced input action exists in the asset and remove the reference if it doesn't.
-            foreach (var actionReference in existingReferences)
+            // "UI" action map has been removed or renamed.
```

Comment:
I do not understand how that could be done since Input Actions are based on strings to begin with? How would you determine when you have a valid reference or how would you now the initial ID? How would you detect which one is valid? Also comes from  https://github.com/Unity-Technologies/InputSystem/pull/1807/files I just moved it around a bit.